src/Sulu/Bundle/TagBundle/Tests/Functional/Trash/TagTrashItemHandlerTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\TagBundle\Tests\Functional\Trash;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sulu\Bundle\TagBundle\Entity\Tag;
+use Sulu\Bundle\TagBundle\Tag\TagInterface;
+use Sulu\Bundle\TagBundle\Tag\TagRepositoryInterface;
+use Sulu\Bundle\TagBundle\Trash\TagTrashItemHandler;
+use Sulu\Bundle\TestBundle\Testing\SuluTestCase;
+
+class TagTrashItemHandlerTest extends SuluTestCase
+{
+    /**
+     * @var TagTrashItemHandler
+     */
+    private $tagTrashItemHandler;
+
+    /**
+     * @var TagRepositoryInterface
+     */
+    private $tagRepository;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    public function setUp(): void
+    {
+        static::purgeDatabase();
+        $this->bootKernelAndSetServices();
+    }
+
+    public function bootKernelAndSetServices(): void
+    {
+        static::bootKernel();
+
+        $this->tagTrashItemHandler = static::getContainer()->get('sulu_tag.tag_trash_item_handler');
+        $this->tagRepository = static::getContainer()->get('sulu.repository.tag');
+        $this->entityManager = static::getEntityManager();
+    }
+
+    public function testStoreAndRestore(): void
+    {
+        /** @var Tag $tag1 */
+        $tag1 = $this->tagRepository->createNew();
+        $tag1->setName('First Tag');
+        $tag1->setCreated(new \DateTime('2020-10-10'));
+        $this->entityManager->persist($tag1);
+
+        $tag2 = $this->tagRepository->createNew();
```

Comment:
This tag isn't used, right? Why did you create it?

Reply:
Ah to test that there are no id collisions when restoring the tag, right? ... nevermind then

Reply:
yes thats right 🙂